[BUG] Cash (Access Violation) when file changes externally over a given threshold [Latest Version Jul-2026]

### Description

Compare 2 or 3 files. Change one of the files externally, possibly extensively as I'm comparing a program output as development evolves. There are 200-400K files and the changes are aplied via regex with a python script. I say this because if the changes are small, or perhaps localized, WinMerge does not seem to crash, but if many lines are changed, possibly dispersed through the file, there seems to be a good chance (10-20%) that WinMerge will crash.

[WinMergeConfig-20260712003534.txt](https://github.com/user-attachments/files/29935030/WinMergeConfig-20260712003534.txt)
[WinMergeCrashInfo-20260712003534.txt](https://github.com/user-attachments/files/29935031/WinMergeCrashInfo-20260712003534.txt)

[WinMergeU.exe.9152.dmp](https://github.com/user-attachments/files/29935035/WinMergeU.exe.9152.dmp)

### Steps to reproduce

1. open win merge
2. compare 2 or 3 files of around 200-400K in size (10-15K lines)
3. With WinMerge comparison still being displayed, apply a chage to one fo the files that changes 50-100 individual lines (regex helps with that so that changes are dispersed through the file)
4. WinMerge will often (10-20% of the times) crash with a segmentation fault

### WinMerge version

2.16.56.2 (2026.05)

### Platform

Windows x64

### Additional information

Example

2026-07-12T00:35:40.049 [INFO] 
  WinMerge Crash Information
  ==========================
  Timestamp:        2026-07-12 00:35:27
  Process ID:       9152
  Thread ID:        13132
  Exception Code:   0xC0000005 (Access Violation)
  Exception Addr:   0x00007FF65A97AE8B
  Module Name:      WinMergeU.exe
  Module Base:      0x00007FF65A700000
  Offset:           0x0027AE8B
  Stack Info:
  ~~~
  SymInit: Symbol-SearchPath: '.;C:\Program Files\WinMerge;C:\Program Files\WinMerge;C:\WINDOWS;C:\WINDOWS\system32;SRV*C:\websymbols*https://msdl.microsoft.com/download/symbols;', symOptions: 530, UserName: 'Ed'
  OS-Version: 10.0.26200 () 0x300-0x1
  C:\Program Files\WinMerge\WinMergeU.exe:WinMergeU.exe (00007FF65A700000), size: 6516736 (result: 0), SymType: '-nosymbols-', PDB: 'C:\Program Files\WinMerge\WinMergeU.exe', fileVersion: 2.16.56.2
  C:\WINDOWS\SYSTEM32\ntdll.dll:ntdll.dll (00007FF8DB000000), size: 2514944 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\ntdll.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\KERNEL32.DLL:KERNEL32.DLL (00007FF8D97E0000), size: 823296 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\KERNEL32.DLL', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\KERNELBASE.dll:KERNELBASE.dll (00007FF8D8640000), size: 4169728 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\KERNELBASE.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\System32\SHLWAPI.dll:SHLWAPI.dll (00007FF8DA960000), size: 421888 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\SHLWAPI.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\ucrtbase.dll:ucrtbase.dll (00007FF8D8410000), size: 1359872 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\ucrtbase.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\IMM32.dll:IMM32.dll (00007FF8DA1D0000), size: 204800 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\IMM32.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\win32u.dll:win32u.dll (00007FF8D8610000), size: 159744 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\win32u.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\System32\USER32.dll:USER32.dll (00007FF8D90B0000), size: 1863680 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\USER32.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\GDI32.dll:GDI32.dll (00007FF8DA750000), size: 176128 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\GDI32.dll', fileVersion: 10.0.26100.8328
  C:\WINDOWS\System32\gdi32full.dll:gdi32full.dll (00007FF8D8C40000), size: 1224704 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\gdi32full.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\msvcp_win.dll:msvcp_win.dll (00007FF8D8560000), size: 667648 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\msvcp_win.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\ADVAPI32.dll:ADVAPI32.dll (00007FF8D8FF0000), size: 745472 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\ADVAPI32.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\msvcrt.dll:msvcrt.dll (00007FF8DA480000), size: 692224 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\msvcrt.dll', fileVersion: 7.0.26100.8246
  C:\WINDOWS\System32\sechost.dll:sechost.dll (00007FF8DA210000), size: 696320 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\sechost.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\RPCRT4.dll:RPCRT4.dll (00007FF8D8D70000), size: 1146880 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\RPCRT4.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\SHELL32.dll:SHELL32.dll (00007FF8D98B0000), size: 7880704 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\SHELL32.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\System32\ole32.dll:ole32.dll (00007FF8DA5B0000), size: 1671168 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\ole32.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.26100.8521_none_3e08b75ce33403b3\COMCTL32.dll:COMCTL32.dll (00007FF8BCCB0000), size: 2695168 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\WinSxS\amd64_microsoft.windows.common-controls_6595b64144ccf1df_6.0.26100.8521_none_3e08b75ce33403b3\COMCTL32.dll', fileVersion: 6.10.26100.8521
  C:\WINDOWS\System32\combase.dll:combase.dll (00007FF8DABF0000), size: 3682304 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\combase.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\UxTheme.dll:UxTheme.dll (00007FF8D5180000), size: 696320 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\UxTheme.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\OLEAUT32.dll:OLEAUT32.dll (00007FF8DA2C0000), size: 888832 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\OLEAUT32.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\WinSxS\amd64_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.26100.8655_none_6ef64fdc3dc4305b\gdiplus.dll:gdiplus.dll (00007FF8BCF50000), size: 1933312 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\WinSxS\amd64_microsoft.windows.gdiplus_6595b64144ccf1df_1.1.26100.8655_none_6ef64fdc3dc4305b\gdiplus.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\OLEACC.dll:OLEACC.dll (00007FF8BD970000), size: 483328 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\OLEACC.dll', fileVersion: 7.2.26100.8521
  C:\WINDOWS\SYSTEM32\icu.dll:icu.dll (00007FF8BBA20000), size: 2752512 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\icu.dll', fileVersion: 72.1.0.4
  C:\WINDOWS\System32\shcore.dll:shcore.dll (00007FF8DA040000), size: 1007616 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\shcore.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\kernel.appcore.dll:kernel.appcore.dll (00007FF8D7020000), size: 110592 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\kernel.appcore.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\System32\bcryptPrimitives.dll:bcryptPrimitives.dll (00007FF8D8B90000), size: 688128 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\bcryptPrimitives.dll', fileVersion: 10.0.26100.8328
  C:\WINDOWS\System32\MSCTF.dll:MSCTF.dll (00007FF8DAA90000), size: 1429504 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\MSCTF.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\mlang.dll:mlang.dll (00007FF88C8C0000), size: 278528 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\mlang.dll', fileVersion: 10.0.26100.7019
  C:\Program Files\WinMerge\darkmodelib\darkmode.dll:darkmode.dll (00007FF8B94A0000), size: 253952 (result: 0), SymType: '-exported-', PDB: 'C:\Program Files\WinMerge\darkmodelib\darkmode.dll', fileVersion: 0.21.2.0
  C:\WINDOWS\SYSTEM32\dwmapi.dll:dwmapi.dll (00007FF8D5430000), size: 192512 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\dwmapi.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\sxs.dll:sxs.dll (00007FF8D80B0000), size: 659456 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\sxs.dll', fileVersion: 10.0.26100.8328
  C:\WINDOWS\SYSTEM32\TextShaping.dll:TextShaping.dll (00007FF89EF60000), size: 733184 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\TextShaping.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\textinputframework.dll:textinputframework.dll (00007FF89EE10000), size: 1343488 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\textinputframework.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\CoreMessaging.dll:CoreMessaging.dll (00007FF8D4C50000), size: 1200128 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\CoreMessaging.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\CoreUIComponents.dll:CoreUIComponents.dll (00007FF8D09E0000), size: 3031040 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\CoreUIComponents.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\wintypes.dll:wintypes.dll (00007FF8D2260000), size: 1482752 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\wintypes.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL:CRYPTBASE.DLL (00007FF8D7890000), size: 49152 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\CRYPTBASE.DLL', fileVersion: 10.0.26100.8246
  C:\WINDOWS\System32\clbcatq.dll:clbcatq.dll (00007FF8D9280000), size: 712704 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\System32\clbcatq.dll', fileVersion: 2001.12.10941.16384
  C:\WINDOWS\system32\dataexchange.dll:dataexchange.dll (00007FF8A53D0000), size: 376832 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\system32\dataexchange.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\system32\twinapi.appcore.dll:twinapi.appcore.dll (00007FF8C8530000), size: 2420736 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\system32\twinapi.appcore.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\Windows.Storage.dll:Windows.Storage.dll (00007FF8D5E40000), size: 8912896 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\Windows.Storage.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\VERSION.dll:VERSION.dll (00007FF8CFE00000), size: 45056 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\VERSION.dll', fileVersion: 10.0.26100.1150
  C:\Program Files\WinMerge\Merge7z\Merge7z.dll:Merge7z.dll (00007FF8B3380000), size: 360448 (result: 0), SymType: '-exported-', PDB: 'C:\Program Files\WinMerge\Merge7z\Merge7z.dll', fileVersion: 26.1.0.0
  C:\WINDOWS\SYSTEM32\PROPSYS.dll:PROPSYS.dll (00007FF8D2AD0000), size: 1073152 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\PROPSYS.dll', fileVersion: 7.0.26100.8521
  C:\WINDOWS\SYSTEM32\CFGMGR32.dll:CFGMGR32.dll (00007FF8D7E40000), size: 331776 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\CFGMGR32.dll', fileVersion: 10.0.26100.8115
  C:\Windows\System32\urlmon.dll:urlmon.dll (00007FF8B9BB0000), size: 1953792 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\urlmon.dll', fileVersion: 11.0.26100.8521
  C:\Windows\System32\iertutil.dll:iertutil.dll (00007FF8B9790000), size: 2904064 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\iertutil.dll', fileVersion: 11.0.26100.8521
  C:\Windows\System32\netutils.dll:netutils.dll (00007FF8D69C0000), size: 53248 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\netutils.dll', fileVersion: 10.0.26100.1882
  C:\Windows\System32\srvcli.dll:srvcli.dll (00007FF8BC7D0000), size: 167936 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\srvcli.dll', fileVersion: 10.0.26100.1150
  C:\Windows\System32\SspiCli.dll:SspiCli.dll (00007FF8D72C0000), size: 299008 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\SspiCli.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\virtdisk.dll:virtdisk.dll (00007FF8BC180000), size: 94208 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\virtdisk.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\profapi.dll:profapi.dll (00007FF8D81B0000), size: 167936 (result: 0), SymType: '-nosymbols-', PDB: 'C:\WINDOWS\SYSTEM32\profapi.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\apphelp.dll:apphelp.dll (00007FF8D4F50000), size: 655360 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\apphelp.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\LINKINFO.dll:LINKINFO.dll (00007FF897F90000), size: 61440 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\LINKINFO.dll', fileVersion: 10.0.26100.1
  C:\WINDOWS\SYSTEM32\ntshrui.dll:ntshrui.dll (00007FF89BBA0000), size: 688128 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\ntshrui.dll', fileVersion: 10.0.26100.8655
  C:\WINDOWS\SYSTEM32\cscapi.dll:cscapi.dll (00007FF89EC30000), size: 77824 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\cscapi.dll', fileVersion: 10.0.26100.1
  C:\WINDOWS\SYSTEM32\policymanager.dll:policymanager.dll (00007FF8D3030000), size: 745472 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\policymanager.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\WindowsCodecs.dll:WindowsCodecs.dll (00007FF8D3890000), size: 2277376 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\WindowsCodecs.dll', fileVersion: 10.0.26100.8655
  C:\Windows\System32\MrmCoreR.dll:MrmCoreR.dll (00007FF8B1950000), size: 1130496 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\MrmCoreR.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\windows.staterepositorycore.dll:windows.staterepositorycore.dll (00007FF8B0C20000), size: 118784 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\windows.staterepositorycore.dll', fileVersion: 10.0.26100.8521
  C:\Windows\System32\thumbcache.dll:thumbcache.dll (00007FF898D90000), size: 454656 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\thumbcache.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\windows.staterepositoryclient.dll:windows.staterepositoryclient.dll (00007FF8B0BD0000), size: 290816 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\windows.staterepositoryclient.dll', fileVersion: 10.0.26100.8521
  C:\Windows\System32\bcp47mrm.dll:bcp47mrm.dll (00007FF8A2160000), size: 212992 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\bcp47mrm.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\edputil.dll:edputil.dll (00007FF8A8B30000), size: 167936 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\edputil.dll', fileVersion: 10.0.26100.8115
  C:\Program Files\Common Files\microsoft shared\ink\tiptsf.dll:tiptsf.dll (00007FF8A8420000), size: 647168 (result: 0), SymType: '-exported-', PDB: 'C:\Program Files\Common Files\microsoft shared\ink\tiptsf.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\UIAutomationCore.DLL:UIAutomationCore.DLL (00007FF8A3A20000), size: 4325376 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\UIAutomationCore.DLL', fileVersion: 7.2.26100.8521
  C:\Windows\System32\Windows.FileExplorer.Common.dll:Windows.FileExplorer.Common.dll (00007FF8A7D10000), size: 786432 (result: 0), SymType: '-exported-', PDB: 'C:\Windows\System32\Windows.FileExplorer.Common.dll', fileVersion: 10.0.26100.8521
  C:\WINDOWS\SYSTEM32\wtdccm.dll:wtdccm.dll (00007FF86D9E0000), size: 69632 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\wtdccm.dll'
  C:\WINDOWS\SYSTEM32\dbghelp.dll:dbghelp.dll (00007FF8D5BC0000), size: 2367488 (result: 0), SymType: '-exported-', PDB: 'C:\WINDOWS\SYSTEM32\dbghelp.dll', fileVersion: 10.0.26100.8328
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A97AE8B)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A97AE8B)
  00007FF65A97AE8B (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A872E18)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A872E18)
  00007FF65A872E18 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A914BFB)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A914BFB)
  00007FF65A914BFB (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A914F7C)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A914F7C)
  00007FF65A914F7C (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A915EEB)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A915EEB)
  00007FF65A915EEB (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A91793E)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A91793E)
  00007FF65A91793E (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A911641)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A911641)
  00007FF65A911641 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A8F9F60)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A8F9F60)
  00007FF65A8F9F60 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A99736C)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A99736C)
  00007FF65A99736C (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A9987D1)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A9987D1)
  00007FF65A9987D1 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A991AC7)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A991AC7)
  00007FF65A991AC7 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A992543)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A992543)
  00007FF65A992543 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8D90BC395)
  00007FF8D90BC395 (USER32): (filename not available): CallWindowProcW
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8D90BBE5B)
  00007FF8D90BBE5B (USER32): (filename not available): CallWindowProcW
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCED7)
  00007FF8BCCDCED7 (COMCTL32): (filename not available): DefSubclassProc
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCE07)
  00007FF8BCCDCE07 (COMCTL32): (filename not available): DefSubclassProc
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8B94AADD1)
  00007FF8B94AADD1 (darkmode): (filename not available): removeWindowEraseBgSubclass
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCED7)
  00007FF8BCCDCED7 (COMCTL32): (filename not available): DefSubclassProc
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCE07)
  00007FF8BCCDCE07 (COMCTL32): (filename not available): DefSubclassProc
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8B94AC7FB)
  00007FF8B94AC7FB (darkmode): (filename not available): removeWindowCtlColorSubclass
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCED7)
  00007FF8BCCDCED7 (COMCTL32): (filename not available): DefSubclassProc
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8BCCDCCAE)
  00007FF8BCCDCCAE (COMCTL32): (filename not available): GetWindowSubclass
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8D90BC395)
  00007FF8D90BC395 (USER32): (filename not available): CallWindowProcW
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8D90BA7EC)
  00007FF8D90BA7EC (USER32): (filename not available): IsWindowUnicode
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A98E501)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A98E501)
  00007FF65A98E501 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A98EE13)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A98EE13)
  00007FF65A98EE13 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65AA31666)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65AA31666)
  00007FF65AA31666 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetSymFromAddr64, GetLastError: 487 (Address: 00007FF65A9E8161)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF65A9E8161)
  00007FF65A9E8161 (WinMergeU): (filename not available): (function-name not available)
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8D980E956)
  00007FF8D980E956 (KERNEL32): (filename not available): BaseThreadInitThunk
  ERROR: SymGetLineFromAddr64, GetLastError: 487 (Address: 00007FF8DB087C1B)
  00007FF8DB087C1B (ntdll): (filename not available): RtlUserThreadStart
  ~~~
  * Crash info: C:\Users\Ed\AppData\Roaming\WinMerge\CrashInfo\WinMergeCrashInfo-20260712003534.txt
  * Crash dump: C:\Users\Ed\AppData\Local\CrashDumps\WinMergeU.exe.9152.dmp
  * Config log: C:\Users\Ed\AppData\Roaming\WinMerge\CrashInfo\WinMergeConfig-20260712003534.txt
2026-07-12T00:35:40.049 [ERROR] WinMerge detected a previous crash. Please report this if possible.

--- a/Src/Common/UniFile.cpp
+++ b/Src/Common/UniFile.cpp
@@ -270,6 +270,14 @@ bool UniMemFile::DoOpen(const String& filename, AccessMode mode)
 	return true;
 }
 
+bool UniMemFile::DoGetFileStatus()
+{
+	m_lastError.ClearError();
+	m_filesize = m_hMapping->end() - m_hMapping->begin();
+	m_statusFetched = 1;
+	return true;
+}
+
 /**
  * @brief Check for Unicode BOM (byte order mark) at start of file
  *

--- a/Src/Common/UniFile.cpp
+++ b/Src/Common/UniFile.cpp
@@ -273,7 +273,18 @@ bool UniMemFile::DoOpen(const String& filename, AccessMode mode)
 bool UniMemFile::DoGetFileStatus()
 {
 	m_lastError.ClearError();
-	m_filesize = m_hMapping->end() - m_hMapping->begin();
+
+	if (m_hMapping == nullptr)
+	{
+		m_filesize = 0;
+		m_statusFetched = 1;
+		return true;
+	}
+
+	auto begin = m_hMapping->begin();
+	auto end = m_hMapping->end();
+
+	m_filesize = end - begin;
 	m_statusFetched = 1;
 	return true;
 }

--- a/Src/Common/UniFile.h
+++ b/Src/Common/UniFile.h
@@ -202,6 +202,7 @@ public:
 // Implementation methods
 protected:
 	virtual bool DoOpen(const String& filename, AccessMode mode);
+	virtual bool DoGetFileStatus();
 
 // Implementation data
 private:

